File: ruby/high-scores/high_scores.rb
class HighScores
  def initialize(scores)
    @scores = scores
  end

  def scores
    @scores
  end

  def latest
    @scores.last
  end

  def highest
    @scores.max
  end

  def top
    @scores.max(3)
  end

  def report
    if latest == highest
      "Your latest score was #{latest}. That's your personal best!"
    elsif latest < highest
      "Your latest score was #{latest}. That's #{highest - latest} short of your personal best!"
    end
  end
end

File: javascript/high-scores/high-scores.js
export class HighScores {
  constructor(scores) {
    this.scores = scores
    this.latest = scores[this.scores.length-1]
    this.highest = Math.max(...this.scores)
    this.top = [...this.scores].sort(this.sortNumber).reverse().slice(0, 3)
  }

  sortNumber(a, b) {
    return a -b;
  }
}
